### ISSUE TITLE
Disappearing chemical cartridges when upgrading

### ISSUE THREAD

**Issue by TrickingTrapster** [ARF-SS13/coyote-bayou#2115]
---- Testmerges: ----
Round ID: 2003
BYOND Version: 514.1589
Server revision compiled on: 2023-04-30T04:01:20+00:00
Master commit: [34b3977c6413222a8459e8fa427680028ebfd007](https://github.com/ARF-SS13/coyote-bayou/commit/34b3977c6413222a8459e8fa427680028ebfd007)
TGS version: 5.2.4
DMAPI version: 5.6.0

---- Reproduction: ----
Use up the starting cartridge for the chem dispenser, then replace it with the same type of cartridge (the knock-off one.) Then, upgrade it - The dispenser will revert to the volume of the original cartridge, essentially emptying the new one with no way of recovery. This does not happen when you use pristine or ancient cartridges, only knock-off ones, but it can waste a TON of resources if it is done in steps. Even if you don't -actually- upgrade the thing, just use the BSRPED to check it, it happens.

---- Screenshots: ----
No screenshots, but easy to reproduce.

---- Additional context: ----
I had a round where 4 entire cartridges were wasted because upgrades kept being checked, and there was no way to get any of the resources back.

**Comment by gozulio** [comment]
I take it this has to do with upgrading using an RPED and not upgrading it by hand?

**Comment by TrickingTrapster** [comment]
Yes, RPED or BSRPED. I have not seen it being upgraded by hand so I don't know if that is the case. It is very consistent with the RPEDs though.

**Comment by TrickingTrapster** [comment]
I just found out it works the other way around too - leave the original cartridge full, mess with upgrading the chem dispenser, and an empty cartridge will be full.
